sopel/modules/url.py

```diff
@@ -185,40 +222,19 @@ def check_callbacks(bot, trigger, url, run=True):
 
 
 def find_title(url, verify=True):
-    """Return the title for the given URL."""
     try:
-        response = requests.get(url, stream=True, verify=verify,
```

Comment:
one key propery of the old code sits right here, with that `stream=True` parameter. I worked on that part and I can tell you i think it's critical to the security of the bot.

because anyone can send the bot to fetch anything, we need to *stop* trying to parse the header after a while, otherwise someone can just continously stream content to the bot and hang that thread completely.